src/crimson/osd/main_config_bootstrap_helpers.cc

```diff
@@ -150,15 +150,16 @@ _get_early_config(int argc, const char *argv[])
 	      std::end(early_args),
 	      [](auto* arg) { return "--smp"sv == arg; });
 	    found == std::end(early_args)) {
-
-	  // Set --smp based on crimson_seastar_smp config option
-	  ret.early_args.emplace_back("--smp");
-
-	  auto smp_config = local_conf().get_val<uint64_t>(
-	    "crimson_seastar_smp");
-
-	  ret.early_args.emplace_back(fmt::format("{}", smp_config));
-	  logger().info("get_early_config: set --smp {}", smp_config);
+	  auto smp_config = crimson::common::get_conf<std::string>("crimson_seastar_cpu_cores");
+	  if (!smp_config.empty()) {
+	    // Set --cpuset based on crimson_seastar_cpu_cores config option
+	    // --smp default is one per CPU
+	    ret.early_args.emplace_back("--cpuset");
+	    ret.early_args.emplace_back(smp_config);
+	    logger().info("get_early_config: set --cpuset {}", smp_config);
+	  } else {
+	    logger().warn("get_early_config: no cpuset, falling back to default");
```

Comment:
What is the "default" in this case?

Reply:
Yeah, best to specify what the fallback actually is.

Reply:
added